tests/plugin_filter/test_plugin_filter.py

```diff
@@ -14,143 +14,316 @@
-    plugin_id = create_or_update_plugin(tmp_db, tags=[tag])
-    plugin = tmp_db.session.query(RAMP).filter(RAMP.id == plugin_id).first()
-    template_tab_id = create_template_tab(
-        tmp_db, client, template_id=template_id, filter_dict={"tag": tag_name}
+
+
+@pytest.fixture(scope="function")
+def plugins(tmp_db):
+    # add plugins generated by hypothesis
+    plugin_names = st.lists(st.text(min_size=1), min_size=1).example()
+    plugin_versions = st.lists(st.from_regex(Version._regex), min_size=1).example()
+    plugin_tag_names = tuple(set(st.lists(st.text(min_size=1), min_size=1).example()))
+    plugin_tags = PluginTag.get_or_create_all(plugin_tag_names)
+
+    for tag in plugin_tags:
+        p_tags = random.sample(plugin_tags, random.randint(0, len(plugin_tags) - 1))
+        if tag not in p_tags:
+            p_tags.append(tag)
+        for name in plugin_names:
+            for version in plugin_versions:
+                plugin_exists = tmp_db.session.query(
+                    exists().where(RAMP.name == name).where(RAMP.version == version)
+                ).scalar()
+                if not plugin_exists:
+                    create_plugin(tmp_db, name=name, version=version, tags=p_tags)
+
+    # add fixed plugins
+    plugin_names = {f"test_plugin_{i}" for i in range(5)}
+    plugin_versions = {"1.0.0", "1.0.1", "1.1.0", "2.0.0"}
+    plugin_tag_names = tuple(f"test_tag_{i}" for i in range(5))
+    plugin_tags = PluginTag.get_or_create_all(plugin_tag_names)
+
+    for name, version, tag in zip(plugin_names, plugin_versions, plugin_tags):
+        plugin_exists = tmp_db.session.query(
+            exists().where(RAMP.name == name).where(RAMP.version == version)
+        ).scalar()
+        if not plugin_exists:
+            create_plugin(tmp_db, name=name, version=version)
+
+    yield tmp_db.session.query(RAMP).all()
+
+
+"""
+Test plugin filtering by name.
+
+The health check is disabled because the test database and client fixtures are not reset between examples generated by `@given(...)`. 
+See https://hypothesis.readthedocs.io/en/latest/healthchecks.html#hypothesis.HealthCheck.function_scoped_fixture for more information.
+"""
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture], deadline=500)
+@given(name=st.text(min_size=1))
+def test_plugin_filter_name(tmp_db, client, template_tab, plugins, name: str):
+    # create plugin
+    plugin_exists = tmp_db.session.query(exists().where(RAMP.name == name)).scalar()
+    if not plugin_exists:
+        create_plugin(tmp_db, name=name)
+
+    # test single name filter
+    tab_plugins = update_plugin_filter(tmp_db, client, template_tab, {"name": name})
+    filtered_plugin_ids = {
+        p_id for p_id, in tmp_db.session.query(RAMP.id).filter(RAMP.name == name).all()
+    }
+    assert len(tab_plugins) > 0
+    assert len(tab_plugins) == len(filtered_plugin_ids)
+    for plugin in tab_plugins:
+        assert plugin.id in filtered_plugin_ids
```

Comment:
you could also convert `tab_plugins` to a plugin id set and then directly compare the sets in a single assert. This would be easier to read and probably also gives better error messages.